metadata/src/main/java/org/apache/kafka/metadata/migration/KRaftMigrationDriver.java

```diff
@@ -222,7 +245,7 @@ private boolean isValidStateChange(MigrationDriverState newState) {
 
     private void transitionTo(MigrationDriverState newState) {
         if (!isValidStateChange(newState)) {
-            log.error("Error transition in migration driver from {} to {}", migrationState, newState);
+            log.error("Invalid transition in migration driver from {} to {}", migrationState, newState);
```

Comment:
I'm confused about this function:

1. Shouldn't the transition to UNINITIALIZED being invalid be covered by isValidStateChange returning false if you pass it newState = UNINITIALIZED ?
2. The switch statement below throws an exception on invalid state change, whereas the initial check just logs a message at `error`. The exception seems more reasonable?
3. the switch has a case for INACTIVE that just does nothing?
```
        switch (newState) {
...
            case INACTIVE:
                // Any state can go to INACTIVE.
                break;
        }

```

Just leave out the case for `INACTIVE` since it's a no-op? (And maybe the whole switch, see above.)

Reply:
(I realize you aren't adding this, but just saw an opportunity for a minor cleanup)

Reply:
sure, i'll clean this up